core/domain/classifier_domain_test.py

```diff
@@ -269,3 +269,59 @@ def test_validation(self):
         with self.assertRaisesRegexp(utils.ValidationError, (
             'Expected training_data to be a list')):
             training_job.validate()
+
+
+class ClassifierExplorationMappingDomainTests(test_utils.GenericTestBase):
+    """Tests for the ClassifierExplorationMapping domain."""
+
+    def _get_mapping_from_dict(self, mapping_dict):
+        mapping = classifier_domain.ClassifierExplorationMapping(
+            mapping_dict['exp_id'],
+            mapping_dict['exp_version'],
+            mapping_dict['state_name'],
+            mapping_dict['classifier_id'])
+
+        return mapping
+
+    def test_to_dict(self):
+        expected_mapping_dict = {
+            'exp_id': 'exp_id1',
+            'exp_version': 2,
+            'state_name': 'state_name1',
+            'classifier_id': 'classifier_id1'
+        }
+        observed_mapping = self._get_mapping_from_dict(
+            expected_mapping_dict)
+        self.assertDictEqual(expected_mapping_dict,
+                             observed_mapping.to_dict())
+
+    def test_validation(self):
```

Comment:
@seanlip . Ahhh. I get it now. So, we will handle it as unicode in the logic layer. That makes sense. I wasn't sure that the config property values would be unicode. Thanks for explaining concisely! So, now I'll encode unicode into str while storing IDs in the storage layer. The rest should just remain unicode. But, still, we don't really need the conversion back. When we encode the state_name to utf-8, we do it locally. So, the ID stored will be a string, but the state_name still remains unicode. My function (get_model) takes exploration parameters and returns the mapping model. So, it internally encodes the unicode state_name to utf-8 and then retrieves the model using that ID. So, currently everything seems well to me. Is there an issue with how I am currently handling conversion?